src/python/pants/init/engine_initializer.py

```diff
@@ -197,6 +197,7 @@ def run_console_rules(self, options_bootstrapper, goals, target_roots):
     subjects = self._determine_subjects(target_roots)
     console = Console()
     # Console rule can only have one subject.
+    # TODO: What about console_rules with no subjects (i.e., no target specs)?
```

Comment:
IIRC, this is now always a single `Specs` object containing a (possibly empty) list of `Spec` objects. Could probably drop/explain the assertion.

Reply:
All I know is, the assert fails when you run with no targets...

Reply:
Well, that's super annoying! Sorry about that.

I can make sure to address that in #6880.